interactions/client/models/command.py

```diff
@@ -181,3 +196,581 @@ class ApplicationCommand(DictSerializerMixin):
     name_localizations: Optional[Dict[Union[str, Locale], str]] = field(default=None)
     description_localizations: Optional[Dict[Union[str, Locale], str]] = field(default=None)
+
+
+def option(
+    _type: OptionType,
+    /,
+    name: str,
+    description: Optional[str] = "No description set",
+    choices: Optional[List[Choice]] = None,
+    required: Optional[bool] = None,
+    channel_types: Optional[List[ChannelType]] = None,
+    min_value: Optional[int] = None,
+    max_value: Optional[int] = None,
+    options: Optional[List[Option]] = None,
+    autocomplete: Optional[bool] = None,
+    focused: Optional[bool] = None,
+    value: Optional[str] = None,
+    name_localizations: Optional[Dict[Union[str, Locale], str]] = None,
+    description_localizations: Optional[Dict[Union[str, Locale], str]] = None,
+) -> Callable[..., Callable[..., Awaitable]]:
+    """
+    A decorator for adding options to a command.
+
+    The structure of an option: ::
+
+        @client.command()
+        @interactions.option(str, name="opt", ...)
+        async def my_command(ctx, opt: str):
+            ...
+
+    :param _type: The type of the option.
+    :type _type: OptionType
+    :param name: The name of the option.
+    :type name: str
+    :param description?: The description of the option. Defaults to ``"No description set"``.
+    :type description: str
+    :param choices?: The choices of the option.
+    :type choices: Optional[List[Choice]]
+    :param required?: Whether the option has to be filled out.
+    :type required: Optional[bool]
+    :param channel_types?: Restrictive shown channel types, if given.
+    :type channel_types: Optional[List[ChannelType]]
+    :param min_value?: The minimum value supported by the option.
+    :type min_value: Optional[int]
+    :param max_value?: The maximum value supported by the option.
+    :type max_value: Optional[int]
+    :param options?: The list of subcommand options included.
+    :type options: Optional[List[Option]]
+    :param autocomplete?: A status denoting whether this option is an autocomplete option.
+    :type autocomplete: Optional[bool]
+    :param focused?: Whether the option is currently being autocompleted or not.
+    :type focused: Optional[bool]
+    :param value?: The value that's currently typed out, if autocompleting.
+    :type value: Optional[str]
+    :param name_localizations?: The dictionary of localization for the ``name`` field. This enforces the same restrictions as the ``name`` field.
+    :type name_localizations: Optional[Dict[Union[str, Locale], str]]
+    :param description_localizations?: The dictionary of localization for the ``description`` field. This enforces the same restrictions as the ``description`` field.
+    :type description_localizations: Optional[Dict[Union[str, Locale], str]]
+    """
+
+    def decorator(coro: Callable[..., Awaitable]) -> Callable[..., Awaitable]:
+        if isinstance(_type, int):
+            type_ = _type
```

Comment:
Why?

Reply:
1. Don't want to use `type` as variable (do u want me to?)
2. I didn't want to reuse the variable ig

Reply:
@EdVraz pls explain

Reply:
changed it to `option_type` and simplified the entire thing